csharp/src/Ice/Internal/IPEndpointI.cs

```diff
@@ -152,8 +150,7 @@ public override bool equivalent(EndpointI endpoint)
         IPEndpointI ipEndpointI = (IPEndpointI)endpoint;
         return ipEndpointI.type() == type() &&
             ipEndpointI.host_.Equals(host_, StringComparison.Ordinal) &&
-            ipEndpointI.port_ == port_ &&
-            Network.addressEquals(ipEndpointI.sourceAddr_, sourceAddr_);
+            ipEndpointI.port_ == port_;
```

Comment:
Removed sourceAddr from this equivalent comparison in C# and Java; in C++, we didn't check sourceAddr.